x-pack/plugins/upgrade_assistant/public/application/components/es_deprecations/deprecations/cell.tsx

```diff
@@ -82,24 +81,38 @@ export const DeprecationCell: FunctionComponent<DeprecationCellProps> = ({
         )}
       </EuiFlexItem>
 
-      {reindexIndexName && (
+      {correctiveAction?.type === 'mlSnapshot' && (
```

Comment:
Since each of these conditions is mutually exclusive, it seems like an appropriate use case for a switch/case statement, possibly extracted into its own component.

```jsx
export const CorrectiveAction ({
  correctiveActionType
}) => {
  switch (correctiveActionType) {
    case 'mlSnapshot':
      return (
        /* ... */
      );

    case 'reindex':
      return (
        /* ... */
      );

    case 'indexSetting':
      return (
        /* ... */
      );

    default:
      throw new Error(`No UI defined for corrective action ${correctiveActionType}`);
  }
};
```